From 5954a5690d566d4f6a264a74b7ea44d4ea496862 Mon Sep 17 00:00:00 2001
From: Marek Goldmann <mgoldman@redhat.com>
Date: Thu, 1 Mar 2012 19:16:29 +0100
Subject: [PATCH 28/43] Added net.jcip, org.jboss.ironjacamar,
 javax.resource.api, org.jboss.as.deployment-scanner,
 com.sun.jsf-impl and many others modules

---
 build/build.xml                                    |  246 ++++++++---------
 build/pom.xml                                      |  290 ++++++++++----------
 .../modules/javax/xml/soap/api/main/module.xml     |    1 -
 pom.xml                                            |    2 +-
 4 files changed, 274 insertions(+), 265 deletions(-)

diff --git a/build/build.xml b/build/build.xml
index cf0b579..56ab4c6 100644
--- a/build/build.xml
+++ b/build/build.xml
@@ -125,6 +125,22 @@
           ~ +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
           -->
 
+        <module-def name="com.sun.jsf-impl">
+            <maven-resource group="com.sun.faces" artifact="jsf-impl"/>
+        </module-def>
+
+        <module-def name="net.jcip">
+            <maven-resource group="net.jcip" artifact="jcip-annotations"/>
+        </module-def>
+
+        <module-def name="javax.inject.api">
+            <maven-resource group="javax.inject" artifact="javax.inject"/>
+        </module-def>
+
+        <module-def name="javax.persistence.api">
+            <maven-resource group="org.hibernate.javax.persistence" artifact="hibernate-jpa-2.0-api"/>
+        </module-def>
+
         <module-def name="org.infinispan">
             <maven-resource group="org.infinispan" artifact="infinispan-core"/>
         </module-def>
@@ -141,16 +157,56 @@
             <maven-resource group="org.infinispan" artifact="infinispan-client-hotrod" />
         </module-def>
 
+        <module-def name="javax.enterprise.api">
+            <maven-resource group="javax.enterprise" artifact="cdi-api"/>
+        </module-def>
+
+        <module-def name="javax.enterprise.deploy.api">
+            <maven-resource group="org.jboss.spec.javax.enterprise.deploy" artifact="jboss-jad-api_1.2_spec"/>
+        </module-def>
+
+        <module-def name="javax.ejb.api">
+            <maven-resource group="org.jboss.spec.javax.ejb" artifact="jboss-ejb-api_3.1_spec"/>
+        </module-def>
+
+        <module-def name="javax.el.api">
+            <maven-resource group="org.jboss.spec.javax.el" artifact="jboss-el-api_2.2_spec"/>
+        </module-def>
+
         <module-def name="javax.annotation.api">
             <maven-resource group="org.jboss.spec.javax.annotation" artifact="jboss-annotations-api_1.1_spec"/>
         </module-def>
 
         <module-def name="javax.api"/>
 
+        <module-def name="javax.faces.api">
+            <maven-resource group="org.jboss.spec.javax.faces" artifact="jboss-jsf-api_2.1_spec"/>
+        </module-def>
+
+        <module-def name="javax.security.auth.message.api">
+            <maven-resource group="org.jboss.spec.javax.security.auth.message" artifact="jboss-jaspi-api_1.0_spec"/>
+        </module-def>
+
+        <module-def name="javax.servlet.jsp.api">
+            <maven-resource group="org.jboss.spec.javax.servlet.jsp" artifact="jboss-jsp-api_2.2_spec"/>
+        </module-def>
+
         <module-def name="javax.interceptor.api">
             <maven-resource group="org.jboss.spec.javax.interceptor" artifact="jboss-interceptors-api_1.1_spec"/>
         </module-def>
 
+        <module-def name="javax.jms.api">
+            <maven-resource group="org.jboss.spec.javax.jms" artifact="jboss-jms-api_1.1_spec"/>
+        </module-def>
+
+        <module-def name="javax.mail.api">
+            <maven-resource group="javax.mail" artifact="mail"/>
+        </module-def>
+
+        <module-def name="javax.resource.api">
+            <maven-resource group="org.jboss.spec.javax.resource" artifact="jboss-connector-api_1.6_spec"/>
+        </module-def>
+
         <module-def name="javax.security.jacc.api">
             <maven-resource group="org.jboss.spec.javax.security.jacc" artifact="jboss-jacc-api_1.4_spec"/>
         </module-def>
@@ -159,24 +215,70 @@
             <maven-resource group="org.jboss.spec.javax.servlet" artifact="jboss-servlet-api_3.0_spec"/>
         </module-def>
 
+        <module-def name="javax.servlet.jstl.api">
+            <maven-resource group="org.jboss.spec.javax.servlet.jstl" artifact="jboss-jstl-api_1.2_spec"/>
+        </module-def>
+
         <module-def name="javax.transaction.api">
             <maven-resource group="org.jboss.spec.javax.transaction" artifact="jboss-transaction-api_1.1_spec"/>
         </module-def>
 
+        <module-def name="javax.xml.bind.api">
+            <maven-resource group="org.jboss.spec.javax.xml.bind" artifact="jboss-jaxb-api_2.2_spec"/>
+        </module-def>
+
         <module-def name="javax.validation.api">
             <maven-resource group="javax.validation" artifact="validation-api"/>
         </module-def>
 
+        <module-def name="javax.xml.soap.api">
+            <maven-resource group="org.jboss.spec.javax.xml.soap" artifact="jboss-saaj-api_1.3_spec"/>
+        </module-def>
+
         <module-def name="javax.xml.stream.api"/>
 
+        <module-def name="javax.xml.rpc.api">
+            <maven-resource group="org.jboss.spec.javax.xml.rpc" artifact="jboss-jaxrpc-api_1.1_spec"/>
+        </module-def>
+
+        <module-def name="javaee.api"/>
+
+        <module-def name="org.apache.commons.logging"/>
+
         <module-def name="org.apache.log4j">
             <maven-resource group="log4j" artifact="log4j"/>
         </module-def>
 
+        <module-def name="org.apache.xalan">
+            <maven-resource group="xalan" artifact="serializer"/>
+            <maven-resource group="xalan" artifact="xalan"/>
+        </module-def>
+
+        <module-def name="org.apache.xerces">
+            <maven-resource group="xerces" artifact="xercesImpl"/>
+        </module-def>
+
         <module-def name="org.hibernate.validator">
             <maven-resource group="org.hibernate" artifact="hibernate-validator"/>
         </module-def>
 
+        <module-def name="org.jboss.ironjacamar.api">
+            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-common-api"/>
+            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-common-spi"/>
+            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-core-api"/>
+        </module-def>
+
+        <module-def name="org.jboss.ironjacamar.impl">
+            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-common-impl"/>
+            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-core-impl"/>
+            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-deployers-common"/>
+            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-validator"/>
+        </module-def>
+
+        <module-def name="org.jboss.ironjacamar.jdbcadapters">
+            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-jdbc"/>
+        </module-def>
+
         <module-def name="org.jboss.as.clustering.api">
             <maven-resource group="org.jboss.as" artifact="jboss-as-clustering-api"/>
         </module-def>
@@ -213,6 +315,10 @@
             <maven-resource group="org.jboss.as" artifact="jboss-as-deployment-repository"/>
         </module-def>
 
+        <module-def name="org.jboss.as.deployment-scanner">
+            <maven-resource group="org.jboss.as" artifact="jboss-as-deployment-scanner"/>
+        </module-def>
+
         <module-def name="org.jboss.as.domain-http-error-context">
             <maven-resource group="org.jboss.as" artifact="jboss-as-domain-http-error-context"/>
         </module-def>
@@ -285,7 +391,7 @@
 
         <module-def name="org.jboss.as.web">
             <maven-resource group="org.jboss.as" artifact="jboss-as-web"/>
-            <!-- <maven-resource group="org.jboss.web" artifact="jasper-jdt"/> -->
+            <maven-resource group="org.eclipse.jdt" artifact="core"/>
             <maven-resource group="org.jboss.web" artifact="jbossweb"/>
         </module-def>
 
@@ -422,6 +528,10 @@
             <maven-resource group="org.jgroups" artifact="jgroups"/>
         </module-def>
 
+        <module-def name="org.joda.time">
+            <maven-resource group="joda-time" artifact="joda-time"/>
+        </module-def>
+
         <module-def name="org.omg.api" />
 
         <module-def name="org.picketbox">
@@ -430,6 +540,18 @@
             <maven-resource group="org.picketbox" artifact="picketbox-commons"/>
         </module-def>
 
+        <module-def name="org.slf4j.jcl-over-slf4j">
+            <maven-resource group="org.slf4j" artifact="jcl-over-slf4j"/>
+        </module-def>
+
+        <module-def name="org.slf4j">
+            <maven-resource group="org.slf4j" artifact="slf4j-api"/>
+        </module-def>
+
+        <module-def name="org.slf4j.impl">
+            <maven-resource group="org.jboss.slf4j" artifact="slf4j-jboss-logmanager"/>
+        </module-def>
+
         <module-def name="sun.jdk"/>
     </target>
 
@@ -462,10 +584,6 @@
             <maven-resource group="com.h2database" artifact="h2"/>
         </module-def>
 
-        <module-def name="com.sun.jsf-impl">
-            <maven-resource group="com.sun.faces" artifact="jsf-impl"/>
-        </module-def>
-
         <module-def name="com.sun.jsf-impl" slot="1.2">
             <maven-resource group="javax.faces" artifact="jsf-impl"/>
         </module-def>
@@ -487,44 +605,14 @@
             <maven-resource group="gnu-getopt" artifact="getopt" />
         </module-def>
 
-        <module-def name="javaee.api"/>
-
         <module-def name="javax.activation.api">
             <maven-resource group="javax.activation" artifact="activation"/>
         </module-def>
 
-        <module-def name="javax.ejb.api">
-            <maven-resource group="org.jboss.spec.javax.ejb" artifact="jboss-ejb-api_3.1_spec"/>
-        </module-def>
-
-        <module-def name="javax.el.api">
-            <maven-resource group="org.jboss.spec.javax.el" artifact="jboss-el-api_2.2_spec"/>
-        </module-def>
-
-        <module-def name="javax.enterprise.api">
-            <maven-resource group="javax.enterprise" artifact="cdi-api"/>
-        </module-def>
-
-        <module-def name="javax.enterprise.deploy.api">
-            <maven-resource group="org.jboss.spec.javax.enterprise.deploy" artifact="jboss-jad-api_1.2_spec"/>
-        </module-def>
-
-        <module-def name="javax.faces.api">
-            <maven-resource group="org.jboss.spec.javax.faces" artifact="jboss-jsf-api_2.1_spec"/>
-        </module-def>
-
         <module-def name="javax.faces.api" slot="1.2">
             <maven-resource group="javax.faces" artifact="jsf-api"/>
         </module-def>
 
-        <module-def name="javax.inject.api">
-            <maven-resource group="javax.inject" artifact="javax.inject"/>
-        </module-def>
-
-        <module-def name="javax.jms.api">
-            <maven-resource group="org.jboss.spec.javax.jms" artifact="jboss-jms-api_1.1_spec"/>
-        </module-def>
-
         <module-def name="javax.jws.api">
             <maven-resource group="javax.jws" artifact="jsr181-api"/>
         </module-def>
@@ -533,56 +621,20 @@
             <maven-resource group="org.jboss.spec.javax.management.j2ee" artifact="jboss-j2eemgmt-api_1.1_spec"/>
         </module-def>
 
-        <module-def name="javax.mail.api">
-            <maven-resource group="javax.mail" artifact="mail"/>
-        </module-def>
-
-        <module-def name="javax.persistence.api">
-            <maven-resource group="org.hibernate.javax.persistence" artifact="hibernate-jpa-2.0-api"/>
-        </module-def>
-
-        <module-def name="javax.resource.api">
-            <maven-resource group="org.jboss.spec.javax.resource" artifact="jboss-connector-api_1.6_spec"/>
-        </module-def>
-
         <module-def name="javax.rmi.api">
             <maven-resource group="org.jboss.spec.javax.rmi" artifact="jboss-rmi-api_1.0_spec"/>
         </module-def>
 
-        <module-def name="javax.security.auth.message.api">
-            <maven-resource group="org.jboss.spec.javax.security.auth.message" artifact="jboss-jaspi-api_1.0_spec"/>
-        </module-def>
-
-        <module-def name="javax.servlet.jsp.api">
-            <maven-resource group="org.jboss.spec.javax.servlet.jsp" artifact="jboss-jsp-api_2.2_spec"/>
-        </module-def>
-
-        <module-def name="javax.servlet.jstl.api">
-            <maven-resource group="org.jboss.spec.javax.servlet.jstl" artifact="jboss-jstl-api_1.2_spec"/>
-        </module-def>
-
         <module-def name="javax.ws.rs.api">
             <maven-resource group="org.jboss.spec.javax.ws.rs" artifact="jboss-jaxrs-api_1.1_spec"/>
         </module-def>
 
-        <module-def name="javax.xml.bind.api">
-            <maven-resource group="org.jboss.spec.javax.xml.bind" artifact="jboss-jaxb-api_2.2_spec"/>
-        </module-def>
-
         <module-def name="javax.xml.jaxp-provider"/>
 
         <module-def name="javax.xml.registry.api">
             <maven-resource group="org.jboss.spec.javax.xml.registry" artifact="jboss-jaxr-api_1.0_spec"/>
         </module-def>
 
-        <module-def name="javax.xml.rpc.api">
-            <maven-resource group="org.jboss.spec.javax.xml.rpc" artifact="jboss-jaxrpc-api_1.1_spec"/>
-        </module-def>
-
-        <module-def name="javax.xml.soap.api">
-            <maven-resource group="org.jboss.spec.javax.xml.soap" artifact="jboss-saaj-api_1.3_spec"/>
-        </module-def>
-
         <module-def name="javax.xml.ws.api">
             <maven-resource group="org.jboss.spec.javax.xml.ws" artifact="jboss-jaxws-api_2.2_spec"/>
         </module-def>
@@ -591,10 +643,6 @@
             <maven-resource group="jline" artifact="jline"/>
         </module-def>
 
-        <module-def name="net.jcip">
-            <maven-resource group="net.jcip" artifact="jcip-annotations"/>
-        </module-def>
-
         <module-def name="net.sourceforge.cssparser">
             <maven-resource group="net.sourceforge.cssparser" artifact="cssparser"/>
         </module-def>
@@ -631,8 +679,6 @@
             <maven-resource group="commons-lang" artifact="commons-lang"/>
         </module-def>
 
-        <module-def name="org.apache.commons.logging"/>
-
         <module-def name="org.apache.commons.pool">
             <maven-resource group="commons-pool" artifact="commons-pool"/>
         </module-def>
@@ -705,15 +751,6 @@
             <maven-resource group="org.apache.juddi.scout" artifact="scout"/>
         </module-def>
 
-        <module-def name="org.apache.xalan">
-            <maven-resource group="xalan" artifact="serializer"/>
-            <maven-resource group="xalan" artifact="xalan"/>
-        </module-def>
-
-        <module-def name="org.apache.xerces">
-            <maven-resource group="xerces" artifact="xercesImpl"/>
-        </module-def>
-
         <module-def name="org.apache.xml-resolver">
             <maven-resource group="xml-resolver" artifact="xml-resolver"/>
         </module-def>
@@ -835,10 +872,6 @@
             <maven-resource group="org.jboss.as" artifact="jboss-as-cmp"/>
         </module-def>
 
-        <module-def name="org.jboss.as.deployment-scanner">
-            <maven-resource group="org.jboss.as" artifact="jboss-as-deployment-scanner"/>
-        </module-def>
-
     	<module-def name="org.jboss.as.domain-add-user" />
 
         <module-def name="org.jboss.as.ejb3">
@@ -972,23 +1005,6 @@
             <maven-resource group="org.jboss.interceptor" artifact="jboss-interceptor-spi"/>
         </module-def>
 
-        <module-def name="org.jboss.ironjacamar.api">
-            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-common-api"/>
-            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-common-spi"/>
-            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-core-api"/>
-        </module-def>
-
-        <module-def name="org.jboss.ironjacamar.impl">
-            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-common-impl"/>
-            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-core-impl"/>
-            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-deployers-common"/>
-            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-validator"/>
-        </module-def>
-
-        <module-def name="org.jboss.ironjacamar.jdbcadapters">
-            <maven-resource group="org.jboss.ironjacamar" artifact="ironjacamar-jdbc"/>
-        </module-def>
-
         <module-def name="org.jboss.jaxbintros">
             <maven-resource group="jboss.jaxbintros" artifact="jboss-jaxb-intros"/>
         </module-def>
@@ -1151,10 +1167,6 @@
             <maven-resource group="org.jboss.jbossts" artifact="jbosstxbridge"/>
         </module-def>
 
-        <module-def name="org.joda.time">
-            <maven-resource group="joda-time" artifact="joda-time"/>
-        </module-def>
-
         <module-def name="org.junit">
             <maven-resource group="junit" artifact="junit"/>
         </module-def>
@@ -1173,22 +1185,10 @@
             <maven-resource group="org.python" artifact="jython-standalone"/>
         </module-def>
 
-        <module-def name="org.slf4j.jcl-over-slf4j">
-            <maven-resource group="org.slf4j" artifact="jcl-over-slf4j"/>
-        </module-def>
-
         <module-def name="org.scannotation.scannotation">
             <maven-resource group="org.scannotation" artifact="scannotation"/>
         </module-def>
 
-        <module-def name="org.slf4j">
-            <maven-resource group="org.slf4j" artifact="slf4j-api"/>
-        </module-def>
-
-        <module-def name="org.slf4j.impl">
-            <maven-resource group="org.jboss.slf4j" artifact="slf4j-jboss-logmanager"/>
-        </module-def>
-
         <module-def name="org.slf4j.ext">
             <maven-resource group="org.slf4j" artifact="slf4j-ext"/>
         </module-def>
diff --git a/build/pom.xml b/build/pom.xml
index c4fd4ce..74aaf52 100644
--- a/build/pom.xml
+++ b/build/pom.xml
@@ -108,6 +108,31 @@
     <dependencies>
 
         <dependency>
+            <groupId>com.sun.faces</groupId>
+            <artifactId>jsf-impl</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>javax.enterprise</groupId>
+            <artifactId>cdi-api</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>javax.inject</groupId>
+            <artifactId>javax.inject</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>javax.mail</groupId>
+            <artifactId>mail</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.hibernate.javax.persistence</groupId>
+            <artifactId>hibernate-jpa-2.0-api</artifactId>
+        </dependency>
+
+        <dependency>
             <groupId>org.infinispan</groupId>
             <artifactId>infinispan-cachestore-jdbc</artifactId>
         </dependency>
@@ -137,6 +162,11 @@
             <artifactId>hibernate-validator</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.eclipse.jdt</groupId>
+            <artifactId>core</artifactId>
+        </dependency>
+
         <!-- TODO: why do we need jandex in minimal? -->
         <dependency>
             <groupId>org.jboss</groupId>
@@ -185,6 +215,16 @@
 
         <dependency>
             <groupId>org.jboss.as</groupId>
+            <artifactId>jboss-as-deployment-repository</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.as</groupId>
+            <artifactId>jboss-as-deployment-scanner</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.as</groupId>
             <artifactId>jboss-as-domain-http-error-context</artifactId>
         </dependency>
 
@@ -369,6 +409,76 @@
         </dependency>
 
         <dependency>
+            <groupId>org.jboss.spec.javax.security.auth.message</groupId>
+            <artifactId>jboss-jaspi-api_1.0_spec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.servlet.jsp</groupId>
+            <artifactId>jboss-jsp-api_2.2_spec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.servlet.jstl</groupId>
+            <artifactId>jboss-jstl-api_1.2_spec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.transaction</groupId>
+            <artifactId>jboss-transaction-api_1.1_spec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.jms</groupId>
+            <artifactId>jboss-jms-api_1.1_spec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.annotation</groupId>
+            <artifactId>jboss-annotations-api_1.1_spec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.xml.rpc</groupId>
+            <artifactId>jboss-jaxrpc-api_1.1_spec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.ejb</groupId>
+            <artifactId>jboss-ejb-api_3.1_spec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.el</groupId>
+            <artifactId>jboss-el-api_2.2_spec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.xml.bind</groupId>
+            <artifactId>jboss-jaxb-api_2.2_spec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.enterprise.deploy</groupId>
+            <artifactId>jboss-jad-api_1.2_spec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.xml.soap</groupId>
+            <artifactId>jboss-saaj-api_1.3_spec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.faces</groupId>
+            <artifactId>jboss-jsf-api_2.1_spec</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.spec.javax.interceptor</groupId>
+            <artifactId>jboss-interceptors-api_1.1_spec</artifactId>
+        </dependency>
+
+        <dependency>
             <groupId>org.jgroups</groupId>
             <artifactId>jgroups</artifactId>
         </dependency>
@@ -387,6 +497,41 @@
             <groupId>org.picketbox</groupId>
             <artifactId>picketbox-infinispan</artifactId>
         </dependency>
+
+        <dependency>
+            <groupId>org.jboss.slf4j</groupId>
+            <artifactId>slf4j-jboss-logmanager</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>joda-time</groupId>
+            <artifactId>joda-time</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>jcl-over-slf4j</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>xalan</groupId>
+            <artifactId>serializer</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>xalan</groupId>
+            <artifactId>xalan</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>xerces</groupId>
+            <artifactId>xercesImpl</artifactId>
+        </dependency>
     </dependencies>
 
     <profiles>
@@ -423,11 +568,6 @@
                 </dependency>
 
                 <dependency>
-                    <groupId>com.sun.faces</groupId>
-                    <artifactId>jsf-impl</artifactId>
-                </dependency>
-
-                <dependency>
                     <groupId>com.sun.xml.bind</groupId>
                     <artifactId>jaxb-impl</artifactId>
                 </dependency>
@@ -500,11 +640,6 @@
                 </dependency>
 
                 <dependency>
-                    <groupId>javax.enterprise</groupId>
-                    <artifactId>cdi-api</artifactId>
-                </dependency>
-
-                <dependency>
                     <groupId>javax.faces</groupId>
                     <artifactId>jsf-api</artifactId>
                 </dependency>
@@ -515,21 +650,11 @@
                 </dependency>
 
                 <dependency>
-                    <groupId>javax.inject</groupId>
-                    <artifactId>javax.inject</artifactId>
-                </dependency>
-
-                <dependency>
                     <groupId>javax.jws</groupId>
                     <artifactId>jsr181-api</artifactId>
                 </dependency>
 
                 <dependency>
-                    <groupId>javax.mail</groupId>
-                    <artifactId>mail</artifactId>
-                </dependency>
-
-                <dependency>
                     <groupId>javax.validation</groupId>
                     <artifactId>validation-api</artifactId>
                 </dependency>
@@ -555,6 +680,11 @@
                 </dependency>
 
                 <dependency>
+                    <groupId>junit</groupId>
+                    <artifactId>junit</artifactId>
+                </dependency>
+
+                <dependency>
                     <groupId>net.jcip</groupId>
                     <artifactId>jcip-annotations</artifactId>
                 </dependency>
@@ -840,11 +970,6 @@
                 </dependency>
 
                 <dependency>
-                    <groupId>org.hibernate.javax.persistence</groupId>
-                    <artifactId>hibernate-jpa-2.0-api</artifactId>
-                </dependency>
-
-                <dependency>
                     <groupId>org.hornetq</groupId>
                     <artifactId>hornetq-core</artifactId>
                 </dependency>
@@ -968,16 +1093,6 @@
 
                 <dependency>
                     <groupId>org.jboss.as</groupId>
-                    <artifactId>jboss-as-deployment-repository</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.as</groupId>
-                    <artifactId>jboss-as-deployment-scanner</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.as</groupId>
                     <artifactId>jboss-as-ejb3</artifactId>
                 </dependency>
 
@@ -1373,46 +1488,6 @@
                 </dependency>
 
                 <dependency>
-                    <groupId>org.jboss.slf4j</groupId>
-                    <artifactId>slf4j-jboss-logmanager</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.spec.javax.annotation</groupId>
-                    <artifactId>jboss-annotations-api_1.1_spec</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.spec.javax.ejb</groupId>
-                    <artifactId>jboss-ejb-api_3.1_spec</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.spec.javax.el</groupId>
-                    <artifactId>jboss-el-api_2.2_spec</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.spec.javax.enterprise.deploy</groupId>
-                    <artifactId>jboss-jad-api_1.2_spec</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.spec.javax.faces</groupId>
-                    <artifactId>jboss-jsf-api_2.1_spec</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.spec.javax.interceptor</groupId>
-                    <artifactId>jboss-interceptors-api_1.1_spec</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.spec.javax.jms</groupId>
-                    <artifactId>jboss-jms-api_1.1_spec</artifactId>
-                </dependency>
-
-                <dependency>
                     <groupId>org.jboss.spec.javax.management.j2ee</groupId>
                     <artifactId>jboss-j2eemgmt-api_1.1_spec</artifactId>
                 </dependency>
@@ -1428,51 +1503,16 @@
                 </dependency>
 
                 <dependency>
-                    <groupId>org.jboss.spec.javax.security.auth.message</groupId>
-                    <artifactId>jboss-jaspi-api_1.0_spec</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.spec.javax.servlet.jsp</groupId>
-                    <artifactId>jboss-jsp-api_2.2_spec</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.spec.javax.servlet.jstl</groupId>
-                    <artifactId>jboss-jstl-api_1.2_spec</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.spec.javax.transaction</groupId>
-                    <artifactId>jboss-transaction-api_1.1_spec</artifactId>
-                </dependency>
-
-                <dependency>
                     <groupId>org.jboss.spec.javax.ws.rs</groupId>
                     <artifactId>jboss-jaxrs-api_1.1_spec</artifactId>
                 </dependency>
 
                 <dependency>
-                    <groupId>org.jboss.spec.javax.xml.bind</groupId>
-                    <artifactId>jboss-jaxb-api_2.2_spec</artifactId>
-                </dependency>
-
-                <dependency>
                     <groupId>org.jboss.spec.javax.xml.registry</groupId>
                     <artifactId>jboss-jaxr-api_1.0_spec</artifactId>
                 </dependency>
 
                 <dependency>
-                    <groupId>org.jboss.spec.javax.xml.rpc</groupId>
-                    <artifactId>jboss-jaxrpc-api_1.1_spec</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.spec.javax.xml.soap</groupId>
-                    <artifactId>jboss-saaj-api_1.3_spec</artifactId>
-                </dependency>
-
-                <dependency>
                     <groupId>org.jboss.spec.javax.xml.ws</groupId>
                     <artifactId>jboss-jaxws-api_2.2_spec</artifactId>
                 </dependency>
@@ -1489,11 +1529,6 @@
 
                 <dependency>
                     <groupId>org.jboss.web</groupId>
-                    <artifactId>jasper-jdt</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.jboss.web</groupId>
                     <artifactId>jbossweb</artifactId>
                 </dependency>
 
@@ -1640,16 +1675,6 @@
 
                 <dependency>
                     <groupId>org.slf4j</groupId>
-                    <artifactId>jcl-over-slf4j</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.slf4j</groupId>
-                    <artifactId>slf4j-api</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>org.slf4j</groupId>
                     <artifactId>slf4j-ext</artifactId>
                 </dependency>
 
@@ -1669,21 +1694,6 @@
                 </dependency>
 
                 <dependency>
-                    <groupId>xalan</groupId>
-                    <artifactId>serializer</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>xalan</groupId>
-                    <artifactId>xalan</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>xerces</groupId>
-                    <artifactId>xercesImpl</artifactId>
-                </dependency>
-
-                <dependency>
                     <groupId>xml-resolver</groupId>
                     <artifactId>xml-resolver</artifactId>
                 </dependency>
diff --git a/build/src/main/resources/modules/javax/xml/soap/api/main/module.xml b/build/src/main/resources/modules/javax/xml/soap/api/main/module.xml
index f65c518..20c547c 100644
--- a/build/src/main/resources/modules/javax/xml/soap/api/main/module.xml
+++ b/build/src/main/resources/modules/javax/xml/soap/api/main/module.xml
@@ -24,7 +24,6 @@
 
 <module xmlns="urn:jboss:module:1.1" name="javax.xml.soap.api">
     <dependencies>
-        <module name="javax.activation.api" export="true"/>
         <module name="javax.api"/>
         <module name="org.jboss.modules"/>
     </dependencies>
diff --git a/pom.xml b/pom.xml
index 6ff1369..8a22682 100644
--- a/pom.xml
+++ b/pom.xml
@@ -277,6 +277,7 @@
         <module>controller</module>
         <module>controller-client</module>
         <module>deployment-repository</module>
+        <module>deployment-scanner</module>
         <module>domain-http</module>
         <module>domain-management</module>
         <module>ee</module>
@@ -5972,7 +5973,6 @@
                 <module>cmp</module>
                 <module>configadmin</module>
                 <module>config-assembly</module>
-                <module>deployment-scanner</module>
                 <module>dist</module>
                 <module>ejb3</module>
                 <module>host-controller</module>
-- 
1.7.10

